lte/gateway/c/oai/lib/mobility_client/MobilityClientAPI.cpp

```diff
@@ -197,101 +191,84 @@ int pgw_handle_allocate_ipv6_address(
           OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNerror);
         }
 
-        memcpy(ip6_addr->s6_addr, ipv6_addr_str.c_str(), sizeof(in6_addr));
+        memcpy(&ip6_addr.s6_addr, ipv6_addr_str.c_str(), sizeof(in6_addr));
         int vlan = atoi(ip_msg.vlan().c_str());
 
-        auto sgi_resp = handle_allocate_ipv6_address_status(
-            status, *ip6_addr, vlan, subscriber_id, apn, pdn_type,
-            sgi_create_endpoint_resp);
-
-        // create session in PCEF and return
-        if (sgi_resp.status == SGI_STATUS_OK) {
-          char ip6_str[INET6_ADDRSTRLEN];
-          inet_ntop(AF_INET6, ip6_addr, ip6_str, INET6_ADDRSTRLEN);
-
-          OAILOG_INFO(
-              LOG_UTIL,
-              "Allocated IPv6 Address <%s>, PDN Type <%s>,"
-              " for IMSI <%s> and APN <%s>\n",
-              ip6_str, pdn_type, subscriber_id, apn);
-
-          s5_create_session_request_t session_req = {0};
-          session_req.context_teid  = sgi_create_endpoint_resp.context_teid;
-          session_req.eps_bearer_id = sgi_create_endpoint_resp.eps_bearer_id;
-          struct pcef_create_session_data session_data;
-          get_session_req_data(
-              spgw_state,
-              &new_bearer_ctxt_info_p->sgw_eps_bearer_context_information
-                   .saved_message,
-              &session_data);
-          pcef_create_session(
-              subscriber_id, nullptr, ip6_str, &session_data, session_req);
-        } else {
-          // If we are here then the IP address allocation has failed
-          s5_response.eps_bearer_id = sgi_create_endpoint_resp.eps_bearer_id;
-          s5_response.context_teid  = sgi_create_endpoint_resp.context_teid;
-          s5_response.failure_cause = IP_ALLOCATION_FAILURE;
-          handle_s5_create_session_response(
-              spgw_state, new_bearer_ctxt_info_p, s5_response);
-        }
-        OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNok);
+        handle_allocate_ipv6_address_status(
+            status, ip6_addr, vlan, subscriber_id, apn, pdn_type, context_teid,
```

Comment:
As with Ipv4 changes, please pass a String object instead of pointer for subscriber_id, apn and pdn_type